quint/src/quintParserFrontend.ts

```diff
@@ -128,6 +129,101 @@ export function parsePhase1(
   }
 }
 
+/**
+ * Phase 1b of the Quint parser. Go over each definition of the form
+ * `import ... from '<path>'`, do the following:
+ * 
+ *  - parse the modules that are refferred by each path,
+ *  - add the parsed modules.
+ * 
+ * Cyclic dependencies among different files are reported as errors.
+ */
+export function parsePhase1b(
+  idGen: IdGenerator,
+  sourceResolver: SourceResolver,
+  mainPath: SourceLookupPath,
+  mainPhase1Result: ParserPhase1
+): ParseResult<ParserPhase1> {
+  // we accumulate the source map over all files here
+  let sourceMap = new Map(mainPhase1Result.sourceMap)
```

Comment:
but why? Anything that has a reference to the map thru this variable will have it's ref mutated anyhow, so why mutate the reference and create a copy of a mutable structure instead of just mutating the mutable structure? Does something at some point keep a reference to the previous map?

Reply:
I do not want to mutate the output of phase1 directly. Phase 1 is called multiple times, once per file. So instead of collecting the source information in the source map of the main file, I am making a copy here.

Reply:
sorry if i'm being dense, but i'm afraid I don't follow: you are already copying the Phase1 output into a map via the `new Map` call, no? Each time you get a new Phase1 result you are adding it to this map. Whether you mutate this map or mutate the ref to add a new map, you are not touching the phase1 result in directly either way, are you?

Reply:
To be clear, this isn't a blocking objection, but i just don't see the benefit to constantly creating copies if you are mutating the state anyhow.

Reply:
> sorry if i'm being dense, but i'm afraid I don't follow: you are already copying the Phase1 output into a map via the `new Map` call, no? Each time you get a new Phase1 result you are adding it to this map. Whether you mutate this map or mutate the ref to add a new map, you are not touching the phase1 result in directly either way, are you?

oh, that's what you mean. This is how I am extending this map:

```js
        sourceMap = new Map([...sourceMap, ...parseResult.value.sourceMap])
```